spec.md

```diff
@@ -1029,6 +1064,153 @@ It is NOT REQUIRED for a controller plugin to implement the `LIST_VOLUMES` capab
 
 ### Node Service RPC
 
+#### `NodeStageVolume`
+
+A Node Plugin MUST implement this RPC call if it has `STAGE_UNSTAGE_VOLUME` node capability.
+
+This RPC is called by the CO prior to the volume being consumed by any workloads on the node by `NodePublishVolume`.
+The Plugin SHALL assume that this RPC will be executed on the node where the volume will be used.
+This RPC SHOULD be called by the CO when a workload that wants to use the specified volume is placed (scheduled) on the specified node for the first time or for the first time since a `NodeUnstageVolume` call for the specified volume was called and returned success on that node.
+
+If the corresponding Controller Plugin has `PUBLISH_UNPUBLISH_VOLUME` controller capability and the Node Plugin has `STAGE_UNSTAGE_VOLUME` capability, then the CO MUST guarantee that this RPC is called after `ControllerPublishVolume` is called for the given volume on the given node and returns a success.
+The CO MUST guarantee that this RPC is called and returns a success before any `NodePublishVolume` is called for the given volume on the given node.
+
+This operation MUST be idempotent.
+If the volume corresponding to the `volume_id` is already staged to the `staging_target_path`, and is identical to the specified `volume_capability` the Plugin MUST reply `0 OK`.
+
+If this RPC failed, or the CO does not know if it failed or not, it MAY choose to call `NodeStageVolume` again, or choose to call `NodeUnstageVolume`.
+
+```protobuf
+message NodeStageVolumeRequest {
+  // The API version assumed by the CO. This is a REQUIRED field.
+  Version version = 1;
+
+  // The ID of the volume to publish. This field is REQUIRED.
+  string volume_id = 2;
+
+  // The CO SHALL set this field to the value returned by
+  // `ControllerPublishVolume` if the corresponding Controller Plugin
+  // has `PUBLISH_UNPUBLISH_VOLUME` controller capability, and SHALL be
+  // left unset if the corresponding Controller Plugin does not have
+  // this capability. This is an OPTIONAL field.
+  map<string, string> publish_volume_info = 3;
+
+  // The path to which the volume will be published. It MUST be an
+  // absolute path in the root filesystem of the process serving this
+  // request. The CO SHALL ensure uniqueness of staging_target_path per
+  // volume.
+  // This is a REQUIRED field.
+  string staging_target_path = 4;
```

Comment:
I get that this falls in line with the publish RPC, but I'm wondering if this is something that really works here.
Since the CO would not consume this path, I'm not sure it's the CO's responsibility to know anything about it.

Reply:
Discussed this on slack, the description of this is really the path to publish for the node, and `NodePublish`'s `target_path` is really the path for the single workload.
This effectively treats the CO as a storage mechanism which I am ok with for this case.

Perhaps the `staging_` should be omitted in the stage RPC?

Reply:
I think having the `staging_` makes it especially clear that this is the same path as the one that is used as `staging_target_path` in `NodePublishVolume` in later calls. Removing it would be naming the same variable differently in different contexts and may introduce some confusion.